tests/cnf/ran/gitopsztp/tests/ztp-cluster-instance-delete.go

```diff
@@ -0,0 +1,182 @@
+
+			if spokeClusterType == ranparam.SNOCluster {
+				Skip("This test only applies to standard or multi-node openshift spoke cluster")
+			}
+
+			// Test step 1-Delete default assisted installer template reference ConfigMap CR after spoke cluster installed.
+
+			// Test step 1 expected result validation.
+			By("verifying installed spoke cluster should still be functional")
+			_, err = version.GetOCPVersion(Spoke1APIClient)
+			Expect(err).ToNot(HaveOccurred(), "Failed to get OCP version from spoke and verify spoke cluster access")
+
+			// Test step 2-Update the ztp-test git path to delete the ClusterInstance CR in root level kustimozation.yaml.
+			By("updating the Argo CD clusters app with the detach AI MNO cluster instance git path")
+			exists, err := gitdetails.UpdateArgoCdAppGitPath(tsparams.ArgoCdClustersAppName,
+				tsparams.ZtpTestPathDetachAIMNO, true)
+			if !exists {
+				Skip(err.Error())
+			}
+
+			Expect(err).ToNot(HaveOccurred(), "Failed to update Argo CD clusters app with new git path")
+
+			// Test step 2 expected results validation.
+			By("checking the infra env manifests removed on hub")
+			_, err = assisted.PullInfraEnvInstall(HubAPIClient, RANConfig.Spoke1Name, RANConfig.Spoke1Name)
+			Expect(err).To(HaveOccurred(), "Found spoke infra env manifests but expected to be removed")
+
+			By("checking the bare metal host manifests removed on hub")
+			_, err = bmh.Pull(HubAPIClient, RANConfig.Spoke1Name, RANConfig.Spoke1Name)
+			Expect(err).To(HaveOccurred(), "Found spoke bmh manifests but expected to be removed")
+
+			By("checking the cluster deployment manifests removed on hub")
+			_, err = hive.PullClusterDeployment(HubAPIClient, RANConfig.Spoke1Name, RANConfig.Spoke1Name)
+			Expect(err).To(HaveOccurred(), "Found spoke cluster deployment manifests but expected to be removed")
+
+			By("checking the NM state config manifests removed on hub")
+			nmStateConfigList, err := assisted.ListNmStateConfigs(HubAPIClient, RANConfig.Spoke1Name)
+			Expect(err).ToNot(HaveOccurred(), "Failed to list NM state config manifests")
+			Expect(nmStateConfigList).To(BeEmpty(), "Found spoke NM state config manifests but expected to be removed")
+
+			By("checking the klusterlet addon config manifests removed on hub")
+			_, err = ocm.PullKAC(HubAPIClient, RANConfig.Spoke1Name, RANConfig.Spoke1Name)
+			Expect(err).To(HaveOccurred(), "Found spoke kac manifests but expected to be removed")
+
+			By("checking the agent cluster install manifests removed on hub")
+			_, err = assisted.PullAgentClusterInstall(HubAPIClient, RANConfig.Spoke1Name, RANConfig.Spoke1Name)
+			Expect(err).To(HaveOccurred(), "Found spoke ACI manifests but expected to be removed")
+
+			By("verifying installed spoke cluster should still be functional")
+			_, err = version.GetOCPVersion(Spoke1APIClient)
+			Expect(err).ToNot(HaveOccurred(), "Failed to get OCP version from spoke and verify spoke cluster access")
```

Comment:
if these are the same between the two cases, they could be moved to a function at the end of this file

Reply:
Okay,updated. Thanks!